custom/icds_reports/ucr/tests/base_test.py

```diff
@@ -60,3 +62,13 @@ def _rebuild_table_get_query_object(self):
         rebuild_indicators(self.datasource._id)
         adapter = IndicatorSqlAdapter(self.datasource)
         return adapter.get_query_object()
+
+    def _run_iterative_monthly_test(self, case_id, cases, start_date=date(2015, 12, 1)):
+        query = self._rebuild_table_get_query_object().filter_by(doc_id=case_id)
+        self.assertEqual(query.count(), 7)
+
+        for index, test_values in cases:
+            row = query.all()[index]._asdict()
+            self.assertEqual(row['month'], start_date+relativedelta(months=index))
+            for key, value in test_values:
+                self.assertEqual(row[key], value)
```

Comment:
The python builtin `enumerate` would make these test cases a bit easier to write (it'd add the indices)
```python
In [1]: for index, val in enumerate(['a', 'b', 'c']):
   ...:     print index, val
   ...:     
0 a
1 b
2 c
```
Or better yet, just drop the `index` business altogether with `zip`:
```python
In [2]: tests = ['test1', 'test2', 'test3']

In [3]: real = ['real1', 'real2', 'real3', 'real4']

In [4]: for test, real in zip(tests, real):
   ...:     print test, real
   ...:     
test1 real1
test2 real2
test3 real3
```